Documentation/proposals/202304-silences.md

```diff
@@ -0,0 +1,112 @@
+# Silence CRD
+
+* Owners:
+  * [mcbenjemaa](https://github.com/mcbenjemaa)
+* Related Tickets:
+  * [#5452](https://github.com/prometheus-operator/prometheus-operator/issues/5452)
+  * [#2398](https://github.com/prometheus-operator/prometheus-operator/issues/2398)
+* Other docs:
+  * n/a
+
+This document describes the creation of `Silence` Custom Resource Definition that defines Alertmanager silences
+configurations in the Kubernetes way.
+
+# Why
+
+prometheus-operator doesn't have a way to automate the management of Alertmanager silences. Many users have either been using some internal scripts
+or an additional operator that does the job e.g. [silence-operator](https://github.com/giantswarm/silence-operator).
+
+* Users who are using CI/CD jobs to manage silences have reported that this is really cumbersome.
+* Users that use scripts are not in better shape obviously.
+* Users that use a standalone operator that exposes a Silence CRD are in a better situation, some folks have reported that they are using GitOps
+  to fully manage the life cycle of Silences Custom resources, Nevertheless, this comes with a drawback
+  as the team must be able to manage a new component in their stack.
+
+Additionally, having a new component in the stack and keeping it maintained is not always ideal (said the folks at [Giant Swarm](https://giantswarm.io) the owners of [silence-operator](https://github.com/giantswarm/silence-operator)),
+because that would require us to fully manage this component as of any managed app (monitoring, logging, alerting etc..).
+Having the Silence CRD as part of prometheus-operator provides users with a better experience: there is no need to install another operator or build a system on top of it to provide that functionality.
```

Comment:
I can see that having 2 components on the stack is more complex than 1, but here we're moving one component by making the other slightly more complex as well, i.e. users still need to monitor Prometheus-operator and Alertmanager to understand if their Silence CRs are behaving as expected.

Of course, I'm saying this without any experience with the silence-operator 😅, it might as well be a super simple component and I'm here making a fuss :P

Could you tell us a bit more about the common failure modes that you've found while using it at Giantswarm?

Reply:
We didn't have complex failures or monitoring for this component, we just need to have an alert that triggers if  a certain Silence reconciliation failed.

Reply:
As @mcbenjemaa said, we have only 2 possible failure scenarios in the silence-operator, either the silence cannot be reconcilied with alertmanager (which almost never happen because we have a small Silence CRD API) or the flux kustomization pulling the silences if failing. 

If this hits into prometheus-operator, the failure modes would roughly be the same in my opinion

Reply:
Exactly